support/ebpf/generic_probe.ebpf.c

```diff
@@ -16,5 +14,19 @@ int uprobe__generic(void *ctx)
 
   u64 ts = bpf_ktime_get_ns();
 
-  return collect_trace(ctx, TRACE_UPROBE, pid, tid, ts, 0);
+  return collect_trace(ctx, TRACE_PROBE, pid, tid, ts, 0);
+}
+
+// uprobe__generic serves as entry point for uprobe based profiling.
+SEC("uprobe/generic")
```

Comment:
Following feedback from https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/651#discussion_r2225256600, we should be able to skip `uprobe/generic` and just use `kprobe/generic` and attach it acccordingly.